src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -20,11 +20,48 @@ internal class GrpcHttpRequestData : HttpRequestData, IAsyncDisposable, IDisposa
         private HttpHeadersCollection? _headers;
         private Stream? _bodyStream;
         private bool _disposed;
+        private readonly Lazy<IReadOnlyCollection<IHttpCookie>> _cookies;
 
         public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             : base(functionContext)
         {
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
+            _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
+            {
+                if(_headers is null)
+                {
+                    return Array.Empty<IHttpCookie>();
+                }
+
+                var cookieString = _headers.FirstOrDefault(item => item.Key == "Cookie").Value;
```

Comment:
Should this be a case insensitive check so that it works for all cases (ex: COOKIE, cookie, Cookie) ? In ASP.NET core, you can read Cookies from HttpContext irrespective of the case of the request header .

Reply:
Or the host is guaranteed to send this it in this case(`Cookie`) irrespective of what the client request header casing is ?

Reply:
FYI: I did a bit of debugging and I see that the cookie header is always received as "Cookie" irrespective of the case sent from the client for the header name.